src/PrestaShopBundle/Controller/Admin/Improve/International/CountryController.php

```diff
@@ -0,0 +1,227 @@
+
+            if ($handlerResult->isSubmitted() && $handlerResult->isValid()) {
+                $this->addFlash('success', $this->trans('Successful creation.', 'Admin.Notifications.Success'));
+
+                return $this->redirectToRoute('admin_countries_index');
+            }
+        } catch (Exception $e) {
+            $this->addFlash('error', $this->getErrorMessageForException($e, $this->getErrorMessages()));
+
+            return $this->redirectToRoute('admin_countries_index');
+        }
+
+        return $this->render('@PrestaShop/Admin/Improve/International/Locations/Country/add.html.twig', [
+            'enableSidebar' => true,
+            'countryForm' => $countryForm->createView(),
+            'help_link' => $this->generateSidebarLink($request->attributes->get('_legacy_controller')),
+            'addressFormat' => $addressFormat->getAddressFormat(),
+            'encodingAddressFormat' => urlencode($addressFormat->getAddressFormat()),
+            'defaultFormat' => urlencode($addressFormat->getDefaultFormat()),
+            'availableFields' => $addressFormat->getAvailableFields(),
+        ]);
+    }
+
+    /**
+     * Handles edit form rendering and submission
+     *
+     * @AdminSecurity(
+     *     "is_granted('update', request.get('_legacy_controller'))",
+     *     redirectRoute="admin_countries_index"
+     * )
+     *
+     * @param int $countryId
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function editAction(int $countryId, Request $request): Response
+    {
+        try {
+            /** @var EditableCountry $editableCountry */
+            $editableCountry = $this->getQueryBus()->handle(new GetCountryForEditing((int) $countryId));
+        } catch (CountryException $e) {
+            $this->addFlash('error', $this->getErrorMessageForException($e, $this->getErrorMessages()));
+
+            return $this->redirectToRoute('admin_countries_index');
+        }
+
+        try {
+            $getAddressFormatDataQuery = new GetAddressFormatData();
+            $getAddressFormatDataQuery->setCountryId($countryId);
+
+            /** @var AddressFormatData $addressFormat */
+            $addressFormat = $this->getQueryBus()->handle($getAddressFormatDataQuery);
+
+            $countryFormBuilder = $this->get(
+                'prestashop.core.form.identifiable_object.builder.country_form_builder'
+            );
+            $countryFormHandler = $this->get(
+                'prestashop.core.form.identifiable_object.handler.country_form_handler'
+            );
+
+            $countryForm = $countryFormBuilder->getFormFor($countryId);
+            $countryForm->handleRequest($request);
+            $result = $countryFormHandler->handleFor($countryId, $countryForm);
+
+            if ($result->isSubmitted() && $result->isValid()) {
+                $this->addFlash('success', $this->trans('Successful update.', 'Admin.Notifications.Success'));
```

Comment:
Decision has been made to use _Update successful_ in _Admin.Notifications.Success_ rather than _Successful update._, cf. https://github.com/PrestaShop/PrestaShop/pull/17819#discussion_r384527574